### PR TITLE
Iox #1078 transform duration inline namespace into normal namespace

### PR DESCRIPTION
**Pull request by elfenpiff**
## Pre-Review Checklist for the PR Author

1. [ ] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [ ] Tests follow the [best practice for testing][testing]
1. [ ] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [ ] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [ ] Commits messages are according to this [guideline][commit-guidelines]
    - [ ] Commit messages have the issue ID (`iox-#123 commit text`)
    - [ ] Commit messages are signed (`git commit -s`)
    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [ ] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [ ] Relevant issues are linked
1. [ ] Add sensible notes for the reviewer
1. [ ] All checks have passed (except `task-list-completed`)
1. [ ] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CHANGELOG.md

## Checklist for the PR Reviewer

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code according to our coding style and naming conventions
- [ ] Unit tests have been written for new behavior
    - [ ] Each unit test case has a unique UUID
- [ ] Public API changes are documented via doxygen
- [ ] Copyright owner are updated in the changed files
- [ ] PR title describes the changes

## Post-review Checklist for the PR Author

1. [ ] All open points are addressed and tracked via issues

## References

- Closes **TBD**

### CODE REVIEW

**Review by elBoberido — APPROVED** [review]

**Comment by elBoberido** [comment]
@elfenpiff don't forget to link the issue and add `Closes #...`

**Comment by elfenpiff** [comment]
@elBoberido thanks for the hint but I am not even sure that this fixes the issue.

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.hpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ enum class TimeSpecReference
 
 class Duration;
 
-inline namespace duration_literals
+namespace duration_literals
```
Comment (mossmaurice) [inline iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.hpp:41]: Please revert this change. The `inline namespace` is needed to make the symbols visible in the outer namespace, see [here](https://stackoverflow.com/a/11018418) for more infos.

**Review by mossmaurice — CHANGES_REQUESTED** [review]

File: iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.hpp (hunk 1 above)
Comment (elfenpiff) [reply iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.hpp:41]: I know what it does but that is something which is not public API and is something which can become dangerous because we implicitly activate this namespace as well.
In general I think we should not use `inline namespaces` at all.

Just think of the problems we encounter when we define other literals as well which are also named `m` like meter for instance and put them into such a inline namespace as well - with have ambiguous symbols.
